src/filedef.cpp

```diff
@@ -1064,7 +1064,12 @@ void FileDefImpl::writeSourceHeader(OutputList &ol)
       ol.endQuickIndices();
     }
     startTitle(ol,getSourceFileBase());
-    ol.parseText(name());
+    QCString nam;
+    if (Config_getBool(FULL_PATH_NAMES))
+      nam = m_docname;
+    else
+      nam = name();
```

Comment:
There were some second thoughts about this construct in https://github.com/albert-github/doxygen/commit/d53f37cfd34af1d51f9715c7e6101fbbd11c3c5b#r117687209.
The solution here should be identical to the solution for #10124